src/care/ProfilePlugin.cpp

```diff
@@ -0,0 +1,63 @@
+#include "care/ProfilePlugin.h"
+#include "care/PluginData.h"
+
+/* CUDA profiling macros */
+#if defined(__CUDACC__) && CARE_HAVE_NVTOOLSEXT
+#include "nvToolsExt.h"
+#endif
+
+namespace care{
+   unsigned int ProfilePlugin::s_current_color = 0;
+   uint32_t ProfilePlugin::s_colors[7] = { 0x0000ff00,   //green
+                                           0x000000ff,   //blue
+                                           0x00ffff00,   //yellow
+                                           0x00ff00ff,   //pink
+                                           0x0000ffff,   //cyan
+                                           0x00ff0000,   //red
+                                           0x00ffffff }; //white
+   int ProfilePlugin::s_num_colors = sizeof(s_colors) / sizeof(uint32_t);
+   bool ProfilePlugin::s_profile_host_loops = true; 
+   
+   void ProfilePlugin::registerPlugin() {
+#if !defined(__HIPCC_)
+      printf("[CARE] Warning: Profiling is only supported in CUDA builds.\n");
```

Comment:
I think you made a typo here in the #if define, unless I am missing something. This should probably be #if defined(__HIPCC__) ? It gives this warning if hip is defined (you have it if hip is not defined I think, with the !, and you deleted one of the underscored after HIPCC)?

Reply:
yes you're right! i will fix that